exe/Main.hs

```diff
@@ -154,7 +154,12 @@ showEvent lock e = withLock lock $ print e
 
 cradleToSession :: Cradle -> IO HscEnvEq
 cradleToSession cradle = do
-    opts <- either throwIO return =<< getCompilerOptions "" cradle
+    result <- getCompilerOptions "" cradle
+    opts <- case result of
+        CradleSuccess x -> return x
+        CradleFail e -> throwIO e
+        CradleNone -> error "Not only did it not work, but there's no error either"
```

Comment:
Looking at the docs at https://github.com/mpickering/hie-bios/blob/8d73b5cb416a1f9b8ecd95d25c094e0f43f0123a/src/HIE/Bios/Cradle.hs#L133, the `none` cradle is intended to be used to ignore a directory so we should probably output something like “Refusing to load directory due to none cradle”. It would also be nice to use `fail` instead of `error` given that we are in `IO` anyway.

Reply:
Done